app/controllers/items_controller.rb

```diff
@@ -0,0 +1,38 @@
+class ItemsController < ApplicationController
+  load_and_authorize_resource :wishlist
+  load_and_authorize_resource :item, :through => :wishlist
+  before_action :authenticate_user!
+  before_action :set_wishlist!
+
+  def new
+    @item = @wishlist.items.new
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_path, alert: t("generic_error")
```

Comment:
I'd redirect to the wishlist page the gift was being added to. If the wishlist is unavailable as well, then the user will be redirected to the index page